src/stackdriver-nozzle/filter/filter.go

```diff
@@ -0,0 +1,50 @@
+package filter
+
+import (
+	"fmt"
+
+	"github.com/cloudfoundry/sonde-go/events"
+	"github.com/evandbrown/gcp-tools-release/src/stackdriver-nozzle/firehose"
+)
+
+type filter struct {
+	dest    firehose.FirehoseHandler
+	enabled map[events.Envelope_EventType]bool
+}
+
+func parseEventName(name string) (events.Envelope_EventType, error) {
+	if eventId, ok := events.Envelope_EventType_value[name]; ok {
+		return events.Envelope_EventType(eventId), nil
+	}
+	return events.Envelope_Error, fmt.Errorf("unknown event name: %s", name)
```

Comment:
Would it make sense to make your own error struct here and return it with the displayed valid events string so the caller can print that?